File: src/src_platform_uwp/fmg/uwp/img/win2d/Mine.cs
using Microsoft.Graphics.Canvas;
using fmg.core.img;

namespace fmg.uwp.img.win2d {

    /// <summary> Mine image on the playing field. Win2D implementation </summary>
    public static class Mine {

        /// <summary> Mine image controller implementation for <see cref="Logo.CanvasBmpView"/> </summary>
        public class CanvasBmpController : Logo.CanvasBmpController {
            public CanvasBmpController(ICanvasResourceCreator resourceCreator)
                : base(resourceCreator)
            {
                LogoModel.ToMineModel(Model);
            }
        }

        /// <summary> Mine image controller implementation for <see cref="Logo.CanvasImgSrcView"/> </summary>
        public class CanvasImgSrcController : Logo.CanvasImgSrcController {
            public CanvasImgSrcController(ICanvasResourceCreator resourceCreator)
                : base(resourceCreator)
            {
                LogoModel.ToMineModel(Model);
            }
        }

    }

}

File: FastMines_android/src/main/java/fmg/android/img/Mine.java
package fmg.android.img;

import fmg.core.img.LogoModel;

/** Mine image on the playing field */
@Deprecated
public final class Mine {
    private Mine() {}

    /** Mine image controller implementation for {@link Logo.BitmapView} */
    public static class BitmapController extends Logo.BitmapController {
        public BitmapController() { LogoModel.toMineModel(getModel()); }
    }

}
